### PR TITLE
[docs] Update Python API docs

### PR DESCRIPTION
**Pull request by marbre**
* Adds a configuration for Read the Docs
* Removes now longer existing content
* Pins requirements to run Sphinx with Python 3.12

### CODE REVIEW

File: docs/api_docs/.readthedocs.yaml
Comment (ScottTodd) [inline docs/api_docs/.readthedocs.yaml:1]: Can you put a link to https://readthedocs.org/projects/iree-python-api/ in `docs/api_docs/README.md`, or here?

I've looked at this directory so many times in passing but never stumbled upon that page that actually shows how the docs are built :P

**Review by ScottTodd — APPROVED** [review]

File: docs/api_docs/.readthedocs.yaml
Comment (marbre) [reply docs/api_docs/.readthedocs.yaml:1]: Sure! I added a link to the README.md and, as this then seems more useful, moved the config to the python subdirectory.

File: docs/api_docs/python/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Documentation for the Python API is built with Sphinx under this directory.
 When new versions are released, the documentation is generated and published
-to the website.
+to [Read the Docs](https://readthedocs.org/projects/iree-python-api/).
```
Comment (ScottTodd) [inline docs/api_docs/python/README.md:5]: Ah, we can include this link _and_ https://iree-python-api.readthedocs.io/en/latest/.

* Users should go to https://iree-python-api.readthedocs.io/en/latest/
* Developers working on the docs can see the configuration, build logs, etc. at https://readthedocs.org/projects/iree-python-api/

Comment (marbre) [reply docs/api_docs/python/README.md:5]: We link to https://iree-python-api.readthedocs.io/en/latest/ at https://github.com/iree-org/iree/blob/main/docs/website/docs/reference/bindings/python.md#usage. So I think it should be fine to link to the config here?

Comment (marbre) [reply docs/api_docs/python/README.md:5]: But it is also fine for me to add a link to https://iree-python-api.readthedocs.io/en/latest/ here. I have some more doc improvements on the list for a follow up anyway.

Comment (ScottTodd) [reply docs/api_docs/python/README.md:5]: Adding more developer docs is cheap and can save future confusion. See the README for https://github.com/iree-org/iree/tree/main/docs/website as an example

**Pull request merged**